src/Bicep.Core/TypeSystem/OciTypeLoader.cs

```diff
@@ -18,7 +23,32 @@ private OciTypeLoader(ImmutableDictionary<string, byte[]> typesCache)
             this.typesCache = typesCache;
         }
 
-        public static OciTypeLoader FromTgz(Stream stream)
+        private static OciTypeLoader FromDiskHelper(IFileSystem fs, Uri? typesTgzUri)
+        {
+            if (typesTgzUri is null)
+            {
+                throw new ArgumentNullException(nameof(typesTgzUri));
+            }
+
+            using var stream = fs.File.OpenRead(typesTgzUri.LocalPath);
+
+            return FromStream(stream);
+        }
+
+        public static OciTypeLoader FromDisk(IFileSystem fs, Uri? typesTgzUri)
+        {
+            try
+            {
+                return FromDiskHelper(fs, typesTgzUri);
```

Comment:
This helper method isn't doing much, and the name isn't clear. Let's just inline the logic here instead:

```csharp
try
{
    using var stream = fs.File.OpenRead(typesTgzUri.LocalPath);

    return FromStream(stream);
    ...
```